addons/web/static/src/js/views/graph/graph_renderer.js

```diff
@@ -1,1079 +1,1049 @@
-     * @param {Object} [params.fields]
-     * @param {string} [params.title]
-     */
-    init: function (parent, state, params) {
-        this._super.apply(this, arguments);
-        this.isEmbedded = params.isEmbedded || false;
-        this.title = params.title || '';
-        this.fields = params.fields || {};
-        this.disableLinking = params.disableLinking;
-
-        this.chart = null;
-        this.chartId = _.uniqueId('chart');
-        this.$legendTooltip = null;
-        this.$tooltip = null;
-    },
-    /**
-     * @override
-     */
-    updateState: function () {
-        return this._super.apply(this, arguments);
-    },
-    /**
-     * Chart.js does not need the canvas to be in dom in order
-     * to be able to work well. We could avoid the calls to on_attach_callback
-     * and on_detach_callback.
-     *
-     * @override
-     */
-    on_attach_callback: function () {
-        this._super.apply(this, arguments);
-        this.isInDOM = true;
-        this._render();
-    },
-    /**
-     * @override
-     */
-    on_detach_callback: function () {
-        this._super.apply(this, arguments);
-        this.isInDOM = false;
-    },
-
-    //--------------------------------------------------------------------------
-    // Private
-    //--------------------------------------------------------------------------
-
-    /**
-     * This function aims to remove a suitable number of lines from the tooltip in order to make it reasonably visible.
-     * A message indicating the number of lines is added if necessary.
-     *
-     * @private
-     * @param {Number} maxTooltipHeight this the max height in pixels of the tooltip
-     */
-    _adjustTooltipHeight: function (maxTooltipHeight) {
-        var sizeOneLine = this.$tooltip.find('tbody tr')[0].clientHeight;
-        var tbodySize = this.$tooltip.find('tbody')[0].clientHeight;
-        var toKeep = Math.floor((maxTooltipHeight - (this.$tooltip[0].clientHeight - tbodySize)) / sizeOneLine) - 1;
-        var $lines = this.$tooltip.find('tbody tr');
-        var toRemove = $lines.length - toKeep;
-        if (toRemove > 0) {
-            $lines.slice(toKeep).remove();
-            var tr = document.createElement('tr');
-            var td = document.createElement('td');
-            tr.classList.add('o_show_more');
-            td.innerHTML = _t("...");
-            tr.appendChild(td);
-            this.$tooltip.find('tbody').append(tr);
+    'use strict';
+
+    const OwlAbstractRenderer = require('web.AbstractRendererOwl');
+    const dataComparisonUtils = require('web.dataComparisonUtils');
+    const config = require('web.config');
+    const DateClasses = dataComparisonUtils.DateClasses;
+    const fieldUtils = require('web.field_utils');
+    const core = require('web.core');
+    const _t = core._t;
+    const qweb = core.qweb;
+
+    const NO_DATA = [_t('No data')];
+    NO_DATA.isNoData = true;
+    const CHART_TYPES = ['pie', 'bar', 'line'];
```

Comment:
no more used apparently

Reply:
Do you have a linter?

Reply:
I'm coding in JS in PyCharm, I guess that's part of my problem.